src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -500,7 +526,7 @@ def update_config(self, save_directory: str):
         with open(config_file_path, "w") as config_file:
             json.dump(config_data, config_file, indent=2, sort_keys=True)
 
-    def _replace_weights(self, dense_weight_generator, model: Module):
+    def _replace_sparsity_weights(self, dense_weight_generator, model: Module):
```

Comment:
Is there any way to combine `_replace_sparsity_weights` and `_replace_weights`? Do they serve any overlapping function?

Reply:
They are separated because of the subtle variations between the sparsity and quantization compressors. The quantization compressor returns not just the decompressed values, but other quantization parameters. The sparsity compressor does not. As these variations are likely to grow, I think it makes sense to keep this functionality separated.